src/common/datetime.c

```diff
@@ -66,7 +66,18 @@ gboolean dt_datetime_exif_to_numbers(dt_datetime_t *dt, const char *exif)
   if(exif && *exif && dt)
   {
     char sdt[DT_DATETIME_LENGTH] = DT_DATETIME_ORIGIN;
-    const int len = strlen(exif) > sizeof(sdt) - 1 ? sizeof(sdt) - 1 : strlen(exif);
+    int len = strlen(exif);
+    // If TZ data is found in the datetime string we should discard it.
+    // We will memcpy this string for parsing and we have to know where to stop so that
+    // the TZ tail after shorter XMP date-time string doesn't damage our parsing buffer.
+    // For possible formats see https://developer.adobe.com/xmp/docs/XMPNamespaces/XMPDataTypes/#date
+    if(exif[len-1] == 'Z')
+      len--;
+    else if(exif[len-3] == '+' || exif[len-3] == '-')
```

Comment:
But if we have an offset number `+/- nn` discarding it means that we are converting to a wrong UTC. No?

Reply:
No, we are not _converting_ as darktable is TZ agnostic (this is a consequence of the fact that most cameras do not know anything about time zones). Most often darktable will deal with EXIF:DateTimeOriginal as the source of date-time information. According to the EXIF standard, the property is in ISO 8601 format but does not include the timezone suffix and should be treated as local time - that is, the time zone in which the photo was taken.

The code discarded TZ information before my fix (the parsing buffer was sized to just fit the date-time string with millisecond precision: "YYYY:mm:dd HH:MM:SS.sss"). Nothing changed here. I only adjusted the length for memcpy because it is possible that the date-time string will have less precision, fewer characters to record the time (for example, only 2 digits of subseconds, or no subseconds at all), but still the TZ tail. In this case, part of the TZ tail could get into the buffer and make it unparsable.

Reply:
Not sure if it's relevant, but there's TimeZoneOffset TIFF (and probably EXIF) tag to specify timezone.

Reply:
Yes, this tag exists. There are also other vendor specific tags for specifying the time zone. So a certain part (not all) of the images may have such information. And yes, it's not relevant.